test/partitioning.jl

```diff
@@ -196,7 +196,31 @@
   end
 
   @testset "SpatialPredicatePartitioner" begin
-    # TODO
+    g = RegularGrid{Float64}(10,10)
+    R = 5.0
+    SPred1(x,y) = norm(x-y) < R
+
+    p = partition(g, SpatialPredicatePartitioner(SPred1))
+    s = subsets(p)
+    testArray = sample(s)
+    @test (norm(sample(testArray)-sample(testArray)) < R) # --> I already know there is an error here. What I'm trying to check is if coordinates of the sampled point agrees with restriction. i.e. I need to get the coordinates of points in RegularGrid. Could you please help me in this assert?
+
+    A = (1,1)
+    B = (10,10)
+    function SPred2(x,y)   # function to check if two points x and y are in a rectangle defined by points A and B
```

Comment:
This function is overly complicated. To check if a point `x` is inside the rectangle defined by corners `A` and `B`, you just need to do `all(A .<= x .<= B)`. So the entire function can be written as

```Julia
# check if points x and y belong to the rectangle [0.,10.]x[0.,10.]
pred(x, y) = all([0.,0.] .<= x .<=[10.,10.]) && all([0.,0.] .<= y .<= [10.,10.])
```

Notice that this may produce a result that is different than what you are expecting. What is the number of subsets you expect to see? Think before you run the code.

Reply:
Thank you. Indeed way more elegant.

This example as it is here should generate just 1 partition. I've commited the wrong version of tuple B. The one I wanted to commit was B=(5,5).
This way, my test should check if there are 65 subsets, with only one of them having more than 1 point (in this case, 36 points exactly).